File: backend/services/post-rust-actix-web-fdm/src/common/api/invalid_title.rs
use crate::common::utils::error::*;

impl super::super::Title {
    pub(in crate::common) fn title_empty() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "TITLE_EMPTY".into(),
                reason: "title is empty".to_string(),
                message: "title is empty".to_string(),
            },
        }
    }
    pub(in crate::common) fn title_too_short() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "TITLE_TOO_SHORT".into(),
                reason: "title is too short".to_string(),
                message: "title is too short".to_string(),
            },
        }
    }
    pub(in crate::common) fn title_too_long() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "TITLE_TOO_LONG".into(),
                reason: "title is too long".to_string(),
                message: "title is too long".to_string(),
            },
        }
    }
}

File: backend/services/post-ktor-mongo-fdm/src/main/kotlin/me/jason5lee/post_ktor_mongo_fdm/common/api/InvalidTitle.kt
package me.jason5lee.post_ktor_mongo_fdm.common.api

import me.jason5lee.post_ktor_mongo_fdm.common.utils.Err
import me.jason5lee.post_ktor_mongo_fdm.common.utils.FailureBody

internal object InvalidTitle {
    internal val empty: FailureBody = FailureBody(
        error = Err(
            error = "TITLE_EMPTY",
            reason = "Title cannot be empty",
            message = "Title cannot be empty",
        )
    )
    internal val tooShort: FailureBody = FailureBody(
        error = Err(
            error = "TITLE_TOO_SHORT",
            reason = "Title must be at least 3 characters",
            message = "Title must be at least 3 characters",
        )
    )
    internal val tooLong: FailureBody = FailureBody(
        error = Err(
            error = "TITLE_TOO_LONG",
            reason = "Title must be at most 20 characters",
            message = "Title must be at most 20 characters",
        )
    )
}